### PR TITLE
feat(charge): create end-to-end tests for the Charge intent

### PR DESCRIPTION
**Pull request by ElliotFriend**
Here's a first effort at some end to end tests in the Charge module. Let me know if it seems comprehensive enough, or if there are more edge-cases we should include?

In the `expect(...).toBe(...)` functions towards the ends of the tests, I opted not to dig into the auth entry credentials too deeply, since that kind of stuff is covered in the unit tests. Should we also test those things here?

### CODE REVIEW

File: sdk/src/charge/e2e.test.ts [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+  }, 30_000)
+
+  it('completes a pull-mode charge', async () => {
+    const serverMppx = MppxServer.create({
+      secretKey: MPP_SECRET_KEY,
+      methods: [
+        serverCharge({
+          recipient: TEST_RECIPIENT,
+          currency: XLM_SAC_TESTNET,
+        }),
```
Comment (marcelosalloum) [inline sdk/src/charge/e2e.test.ts]: After this PR was opened, #42 made `store: Store.Store` a required field on `charge.Parameters` – the factory now throws immediately if none is provided

The four serverCharge({...}) calls here need to be updated to include it, and Store needs to be added to the mppx import.

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Added
+
+- End-toEnd tests for the charge intent, submitting real transactions on Testnet, and making sure the flows work in combinations of push/pull and sponsorship ([#41](https://github.com/stellar/stellar-mpp-sdk/pull/41))
```
Comment (marcelosalloum) [inline CHANGELOG.md]:
```suggestion
- End-to-End tests for the charge intent, submitting real transactions on Testnet, and making sure the flows work in combinations of push/pull and sponsorship ([#41](https://github.com/stellar/stellar-mpp-sdk/pull/41))
```

File: sdk/src/charge/e2e.test.ts [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+    expect(receipt.status).toBe('success')
+    expect(receipt.method).toBe('stellar')
+    expect(receipt.reference).toBeTruthy()
+    // The reference should be a 64-char hex tx hash
+    expect(receipt.reference).toMatch(/^[a-f0-9]{64}$/)
+
+    const txHash = receipt.reference
+    const tx = (await sorobanServer.getTransaction(txHash)) as Api.GetSuccessfulTransactionResponse
+    // The transaction should be successful
+    expect(tx.txHash).toEqual(receipt.reference) // maybe unnecessary?
```
Comment (marcelosalloum) [inline sdk/src/charge/e2e.test.ts]: We can keep it

```suggestion
    expect(tx.txHash).toEqual(receipt.reference)
```

**Review by marcelosalloum — COMMENTED** [review]
Thanks for the PR! A few things to fix before merging:

**Comment by marcelosalloum** [comment]
@claude address the PR comments and make sure CI passes

File: sdk/src/charge/integration/e2e.test.ts [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+    const request = new Request(input, init)
+    const result = await handler(request)
+    if (result.status === 402) {
+      if (!result.challenge) {
+        throw new Error(`handlerAsFetch: 402 result missing challenge: ${JSON.stringify(result)}`)
+      }
+      return result.challenge
+    }
+    if (result.withReceipt) {
+      return result.withReceipt(Response.json({ message: 'paid' }))
```
[the remark this answers is not in the document]
Comment (marcelosalloum) [reply sdk/src/charge/integration/e2e.test.ts]: Addressed in 120c199673cf5aa007d86ad09ebb980fb9fa0609

File: sdk/src/charge/integration/e2e.test.ts [hunk 5]
```diff
@@ -0,0 +1,304 @@
...
+  const clientMppx = MppxClient.create({
+    polyfill: false,
+    fetch: handlerAsFetch(handler),
+    methods: [opts.clientMethod],
+  })
+
+  const response = await clientMppx.fetch('http://localhost/test')
+  expect(response.status).toBe(200)
+
+  const receipt = Receipt.deserialize(response.headers.get('Payment-Receipt')!)
```
[the remark this answers is not in the document]
Comment (marcelosalloum) [reply sdk/src/charge/integration/e2e.test.ts]: Addressed in 120c199673cf5aa007d86ad09ebb980fb9fa0609

File: Makefile [hunk 6]
```diff
@@ -1,5 +1,5 @@
-.PHONY: install build clean typecheck lint lint-fix format format-check test test-watch check \
-        demo-server demo-client demo-channel-server demo-channel-client help
+.PHONY: install build clean typecheck lint lint-fix format format-check test test-watch \
+        test-integration check demo-server demo-client demo-channel-server demo-channel-client help
```
[the remark this answers is not in the document]
Comment (marcelosalloum) [reply Makefile]: Addressed in 120c199673cf5aa007d86ad09ebb980fb9fa0609

File: sdk/src/charge/integration/e2e.test.ts [hunk 7]
```diff
@@ -0,0 +1,304 @@
...
+// All flows share TEST_PAYER so we only fund one payer account per suite run.
+// Consequence: tests are NOT fully independent — the payer's sequence advances
+// between flows, and a tx failure mid-broadcast can surface as a sequence or
+// balance error in a later flow. If a test fails unexpectedly, re-run the
+// whole file rather than isolating a single `it()`.
+const TEST_PAYER = Keypair.random()
+const TEST_RECIPIENT = Keypair.random().publicKey()
+const TEST_ENVELOPE_SIGNER = Keypair.random()
+const TEST_FEE_PAYER = Keypair.random()
+
```
[the remark this answers is not in the document]
Comment (marcelosalloum) [reply sdk/src/charge/integration/e2e.test.ts:39]: That's per design.

**Review by marcelosalloum — APPROVED** [review]
